### PR TITLE
CAB-3812: Adds page objects and E2E test for filter select with multiple selection.

### PR DESCRIPTION
**Pull request by farmas**
Sadly, this changes includes formatting updates from prettier. They get changed at commit time, so its hard to know what's gonna happen.

### CODE REVIEW

File: e2e/page-objects/select-field.po.ts [hunk 1]
```diff
@@ -49,7 +52,7 @@ export class SelectFieldPageObject {
   /**
    * Gets all the dropdown options.
    */
-  getOptions(): ElementArrayFinder {
+  get options(): ElementArrayFinder {
```
Comment (farmas) [inline e2e/page-objects/select-field.po.ts:55]: I changed this to match the new PO. In general, the pattern that I am trying to use is fields/properties to access 'Finders' (this seems to be the preferred approach from protractor because it delays finding the element until its used) and methods for operations that require WebDriver to acquire element references.

File: e2e/edit/edit.local-spec.ts [hunk 2]
```diff
@@ -248,6 +248,25 @@ describe('Edit Page for Demo', () => {
 
     expect(page.getSnackBarText()).toEqual('There was an error deleting content item:Forbidden\nDismiss');
   });
+
+  it('should be able to add and remove options from filter select', () => {
```
Comment (huiaic) [inline e2e/edit/edit.local-spec.ts:252]:
```suggestion
  it('should be able to add and remove options from filter select with multi select', () => {
```

File: e2e/edit/edit.local-spec.ts [hunk 3]
```diff
@@ -248,6 +248,25 @@ describe('Edit Page for Demo', () => {
...
+    filterSelect.waitForSelectionCount(1);
+    expect(filterSelect.getSelectedValues()).toEqual(['val 3 label']);
+    expect(page.saveButton.isEnabled()).toBeTruthy();
+
+    // add an option, verify selected options changed
+    filterSelect.addOptionAt(1);
+    filterSelect.waitForSelectionCount(2);
+    expect(filterSelect.getSelectedValues()).toEqual(['val 3 label', 'val 2 label']);
+  });
 });
```
Comment (huiaic) [inline e2e/edit/edit.local-spec.ts:271]: Maybe add a similar test to Create page to test that new values can added when the field is blank.

Comment (farmas) [reply e2e/edit/edit.local-spec.ts:271]: I decided to add the test for the filter-select that starts empty to the bulkeditpage instead, since that page has special handling for the emptyFormValidator. I did however add the filter-select field to the create page in the demo.json profile.
